data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/http/WebClientCloudEventSender.java

```diff
@@ -135,10 +135,6 @@ private Future<HttpResponse<Buffer>> send(final CloudEvent event, final int retr
           }
         }
 
-        if (retryCounter < consumerVerticleContext.getEgressConfig().getRetry()) {
-          return retry(retryCounter, event);
-        }
```

Comment:
This is handling the case where we have a failure but without a response, I think we're missing a return at line 135 when this `if (isRetryableStatusCode(response.statusCode()) && retryCounter < consumerVerticleContext.getEgressConfig().getRetry())` is false